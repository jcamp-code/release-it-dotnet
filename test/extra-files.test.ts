import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'

import {
  knownDotNetFiles,
  processExtraFiles,
  type ExtraFile,
} from '../src/extra-files'

const fixturesPath = './test/updaters/fixtures'

describe('extra-files', async () => {
  it('updates all files in list', async () => {
    const extraFiles: ExtraFile[] = [
      resolve(fixturesPath, './Version.java'),
      {
        path: resolve(fixturesPath, './helm/Chart.yaml'),
        type: 'yaml',
        jsonpath: '$.version',
      },
      {
        path: resolve(fixturesPath, './Foo.csproj'),
        type: 'xml',
        xpath: '//Project/PropertyGroup/Version',
      },
    ]
    const updates = await processExtraFiles(extraFiles, '3.2.7', true)
    expect(updates[0].newContent).toMatchSnapshot()
    expect(updates[1].newContent).toMatchSnapshot()
    expect(updates[2].newContent).toMatchSnapshot()
  })
  it('updates all known files in list', async () => {
    const extraFiles: ExtraFile[] = [
      resolve(fixturesPath, './Foo.csproj'),
      resolve(fixturesPath, './Directory.Build.props'),
    ]
    const updates = await processExtraFiles(
      extraFiles,
      '3.2.7',
      true,
      knownDotNetFiles,
    )
    expect(updates[0].newContent).toMatchSnapshot()
    expect(updates[1].newContent).toMatchSnapshot()
  })
  it('supports globs of known files', async () => {
    const extraFiles: ExtraFile[] = ['./test/updaters/fixtures/*.csproj']
    const updates = await processExtraFiles(
      extraFiles,
      '3.2.7',
      true,
      knownDotNetFiles,
    )
    expect(updates[0].newContent).toMatchSnapshot()
    expect(updates[1].newContent).toMatchSnapshot()
  })
})
