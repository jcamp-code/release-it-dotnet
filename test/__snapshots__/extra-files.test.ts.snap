// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extra-files > supports globs of known files 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<Project Sdk=\\"Microsoft.NET.Sdk\\">
  <PropertyGroup>
    <Version>3.2.7</Version>
    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <Description>Recommended Google client library to access the Google Cloud Channel API, which enables Google Cloud resellers and distributors to manage their customers, channel partners, entitlements and reports.</Description>
    <PackageTags>channel;reseller;Google;Cloud</PackageTags>
    <CodeAnalysisRuleSet>..\\\\..\\\\..\\\\grpc.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=\\"ConfigureAwaitChecker.Analyzer\\" Version=\\"5.0.0\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Google.Api.Gax.Grpc.GrpcCore\\" Version=\\"[3.5.0, 4.0.0)\\"/>
    <PackageReference Include=\\"Google.LongRunning\\" Version=\\"[2.3.0, 3.0.0)\\"/>
    <PackageReference Include=\\"Grpc.Core\\" Version=\\"[2.38.1, 3.0.0)\\" PrivateAssets=\\"None\\"/>
    <PackageReference Include=\\"Microsoft.DotNet.Analyzers.Compatibility\\" Version=\\"0.2.12-alpha\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Microsoft.NETFramework.ReferenceAssemblies\\" Version=\\"1.0.2\\" PrivateAssets=\\"All\\"/>
    <Analyzer Condition=\\"Exists('..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll')\\" Include=\\"..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll\\"/>
  </ItemGroup>
</Project>"
`;

exports[`extra-files > supports globs of known files 2`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<Project Sdk=\\"Microsoft.NET.Sdk\\">
  <PropertyGroup>
    <Version>3.2.7</Version>
    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <Description>Recommended Google client library to access the Google Cloud Channel API, which enables Google Cloud resellers and distributors to manage their customers, channel partners, entitlements and reports.</Description>
    <PackageTags>channel;reseller;Google;Cloud</PackageTags>
    <CodeAnalysisRuleSet>..\\\\..\\\\..\\\\grpc.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=\\"ConfigureAwaitChecker.Analyzer\\" Version=\\"5.0.0\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Google.Api.Gax.Grpc.GrpcCore\\" Version=\\"[3.5.0, 4.0.0)\\"/>
    <PackageReference Include=\\"Google.LongRunning\\" Version=\\"[2.3.0, 3.0.0)\\"/>
    <PackageReference Include=\\"Grpc.Core\\" Version=\\"[2.38.1, 3.0.0)\\" PrivateAssets=\\"None\\"/>
    <PackageReference Include=\\"Microsoft.DotNet.Analyzers.Compatibility\\" Version=\\"0.2.12-alpha\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Microsoft.NETFramework.ReferenceAssemblies\\" Version=\\"1.0.2\\" PrivateAssets=\\"All\\"/>
    <Analyzer Condition=\\"Exists('..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll')\\" Include=\\"..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll\\"/>
  </ItemGroup>
</Project>"
`;

exports[`extra-files > updates all files in list 1`] = `
"/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the \\"License\\");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an \\"AS IS\\" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.google.cloud.example;

public final class Version {
  // {x-release-please-start-version}
  public static String VERSION = \\"3.2.7\\";
  // {x-release-please-end}

  // {x-release-please-start-major}
  public static String MAJOR = \\"3\\";
  // {x-release-please-end}

  // {x-release-please-start-minor}
  public static String MINOR = \\"2\\";
  // {x-release-please-end}

  // {x-release-please-start-patch}
  public static String PATCH = \\"7\\";
  // {x-release-please-end}

  public static String INLINE_VERSION = \\"3.2.7\\"; // {x-release-please-version}
  public static String INLINE_MAJOR = \\"3\\"; // {x-release-please-major}
  public static String INLINE_MINOR = \\"2\\"; // {x-release-please-minor}
  public static String INLINE_PATCH = \\"7\\"; // {x-release-please-patch}
}
"
`;

exports[`extra-files > updates all files in list 2`] = `
"name: helm-test-repo
version: 3.2.7
apiVersion: v2
appVersion: 2.0.0
dependencies:
  - name: another-repo
    version: 0.15.3
    repository: linkToHelmChartRepo
maintainers:
  - Abhinav Khanna
"
`;

exports[`extra-files > updates all files in list 3`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<Project Sdk=\\"Microsoft.NET.Sdk\\">
  <PropertyGroup>
    <Version>3.2.7</Version>
    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <Description>Recommended Google client library to access the Google Cloud Channel API, which enables Google Cloud resellers and distributors to manage their customers, channel partners, entitlements and reports.</Description>
    <PackageTags>channel;reseller;Google;Cloud</PackageTags>
    <CodeAnalysisRuleSet>..\\\\..\\\\..\\\\grpc.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=\\"ConfigureAwaitChecker.Analyzer\\" Version=\\"5.0.0\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Google.Api.Gax.Grpc.GrpcCore\\" Version=\\"[3.5.0, 4.0.0)\\"/>
    <PackageReference Include=\\"Google.LongRunning\\" Version=\\"[2.3.0, 3.0.0)\\"/>
    <PackageReference Include=\\"Grpc.Core\\" Version=\\"[2.38.1, 3.0.0)\\" PrivateAssets=\\"None\\"/>
    <PackageReference Include=\\"Microsoft.DotNet.Analyzers.Compatibility\\" Version=\\"0.2.12-alpha\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Microsoft.NETFramework.ReferenceAssemblies\\" Version=\\"1.0.2\\" PrivateAssets=\\"All\\"/>
    <Analyzer Condition=\\"Exists('..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll')\\" Include=\\"..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll\\"/>
  </ItemGroup>
</Project>"
`;

exports[`extra-files > updates all known files in list 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<Project Sdk=\\"Microsoft.NET.Sdk\\">
  <PropertyGroup>
    <Version>3.2.7</Version>
    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <Description>Recommended Google client library to access the Google Cloud Channel API, which enables Google Cloud resellers and distributors to manage their customers, channel partners, entitlements and reports.</Description>
    <PackageTags>channel;reseller;Google;Cloud</PackageTags>
    <CodeAnalysisRuleSet>..\\\\..\\\\..\\\\grpc.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=\\"ConfigureAwaitChecker.Analyzer\\" Version=\\"5.0.0\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Google.Api.Gax.Grpc.GrpcCore\\" Version=\\"[3.5.0, 4.0.0)\\"/>
    <PackageReference Include=\\"Google.LongRunning\\" Version=\\"[2.3.0, 3.0.0)\\"/>
    <PackageReference Include=\\"Grpc.Core\\" Version=\\"[2.38.1, 3.0.0)\\" PrivateAssets=\\"None\\"/>
    <PackageReference Include=\\"Microsoft.DotNet.Analyzers.Compatibility\\" Version=\\"0.2.12-alpha\\" PrivateAssets=\\"All\\"/>
    <PackageReference Include=\\"Microsoft.NETFramework.ReferenceAssemblies\\" Version=\\"1.0.2\\" PrivateAssets=\\"All\\"/>
    <Analyzer Condition=\\"Exists('..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll')\\" Include=\\"..\\\\..\\\\..\\\\tools\\\\Google.Cloud.Tools.Analyzers\\\\bin\\\\$(Configuration)\\\\netstandard1.3\\\\publish\\\\Google.Cloud.Tools.Analyzers.dll\\"/>
  </ItemGroup>
</Project>"
`;

exports[`extra-files > updates all known files in list 2`] = `
"<Project>

  <PropertyGroup>
    <RootNamespace>MongoFramework.AspNetCore.Identity</RootNamespace>
    <PackageId>$(AssemblyName)</PackageId>
    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
    <PackageLicenseFile>LICENSE</PackageLicenseFile>
    <PackageProjectUrl>https://github.com/JohnCampionJr/MongoFramework.AspNetCore.Identity</PackageProjectUrl>
    <PackageBaseTags>mongodb;mongoframework;mongo;identity</PackageBaseTags>

    <RepositoryUrl>https://github.com/jcamp-code/release-it</RepositoryUrl>
    <RepositoryType>git</RepositoryType>
    <RepositoryBranch>main</RepositoryBranch>

    <!-- SourceLink Support -->
    <PublishRepositoryUrl>true</PublishRepositoryUrl>
    <EmbedUntrackedSources>true</EmbedUntrackedSources>
    <IncludeSymbols>true</IncludeSymbols>
    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
    <LangVersion>Latest</LangVersion>
    <Version>3.2.7</Version>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include=\\"Microsoft.SourceLink.GitHub\\" Version=\\"1.1.1\\" PrivateAssets=\\"All\\"/>
  </ItemGroup>

  <ItemGroup>
    <None Include=\\"..\\\\..\\\\LICENSE\\" Pack=\\"true\\" Visible=\\"false\\" PackagePath=\\"\\"/>
    <None Include=\\"..\\\\..\\\\README.md\\" Pack=\\"true\\" Visible=\\"false\\" PackagePath=\\"\\"/>
  </ItemGroup>

</Project>
"
`;
